lib/matplotlib/widgets.py

```diff
@@ -600,12 +607,29 @@ def set_active(self, index):
         for cid, func in self.observers.items():
             func(self.labels[index].get_text())
 
+    def clear(self):
```

Comment:
set_active, the way it is implemented, is a toggle function, and it does a few more things that are not necessary for `clear()`. Moreover, clear() is indiscriminate, and hence the current implementation is the most efficient way of achieving it. . because it is clear operation, I don't expect any callbacks/observers to be triggered, unless we offer the option of a callback_on_clear - but I don't think that feature is useful or desirable.